tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/ScalaJSClassEmitter.scala

```diff
@@ -232,7 +273,15 @@ private[scalajs] final class ScalaJSClassEmitter(
             List(js.This()))
       }
       val fieldDefs = genFieldDefs(tree)
-      js.Function(Nil, js.Block(superCtorCall :: fieldDefs))
+      if (!usesJSConstructorOpt(className)) {
+        js.Function(Nil, js.Block(superCtorCall :: fieldDefs))
+      } else {
+        // only handling the case for one constructor so far
+        val (ctor :: Nil) =
```

Comment:
Use `val List(ctor) =` instead. That's the variant we use in this codebase. (It's actually funny; it took me a full minute to understand what was bothering me in this line ^^)